### DISCUSSION TITLE
About Simulation

### DISCUSSION THREAD

**Discussion by prabin2054** [WEC-Sim/WEC-Sim#1337] [Q&A] [answered]
I read the paper and watched the video about the WEC-Sim. 
I installed the Wec-Sim in the library section as well. Even I download RM3 model from WEC-Sim/WEC-Sim.
But I didnot understand how to run the file. 

Error:
WEC-Sim Read Input File ...   
Unrecognized property 'height' for class 'waveClass'.

aany idea? or video?

**Comment by MShabara** [comment, marked as answer]
@prabin2054 
I believe the issue you're facing is not related to the WEC-Sim source code but rather with your WEC-Sim setup. This would be better suited for the discussions tab.

In response to your questions:

> I read the paper and watched the video about the WEC-Sim.

Which paper and which video are you referring to? Please provide more context.
[This video](https://www.youtube.com/watch?v=HlIhYOVCAWA) provides details about installing WEC-Sim.

> I installed WEC-Sim in the library section as well. I even downloaded the RM3 model from WEC-Sim/WEC-Sim.

To install WEC-Sim, please follow these steps:

Download WEC-Sim as shown in the figure below:
![image](https://github.com/user-attachments/assets/5bcb8029-4b69-487d-a1bf-dae381bebd54)

Navigate to the MATLAB `startup.m` directory, example `C:\Users\user\Documents\MATLAB` and open the `startup.m` file.

Add the path to the WEC-Sim source folder. For example, use this line:
`wecSimSource = 'C:\Users\user\Downloads\WEC-Sim\source';`

**Reply by prabin2054** [reply]
I watched that video. But that video describe based on pdf. It would be great, if you show  how you download WEC-Sim, download all files and run that files. So views can understand more quickly. 
Still i didnot understan how to give input on RM3. How to Connect RM3 with code.

**Comment by MShabara** [comment]
WEC-Sim is intended for users who are familiar with MATLAB and Simulink libraries. 

The video mentioned both installation options of WECSim in the second 0:51. 

**To answer your question:**
1. Before running the RM3, please make sure that WEC-Sim is installed properly. Open MATLAB and type `wecSim` in the command window. WEC-Sim should start in the background and search for the input file, and you should then get the following error messages:

<img src="https://github.com/user-attachments/assets/9906903c-c670-4f76-9b24-885728c2acc2" width="400" height="200">


2. To run the RM3 example, navigate to the [RM3 example directory ](https://github.com/WEC-Sim/WEC-Sim/tree/main/examples/RM3) then open the `wecSimInputFile.m`

To run the RM3 model you need to type `wecSim` in the command window then press enter.
